File: Lab4_Android/app/src/main/java/com/example/lab4/MainActivity.java
package com.example.lab4;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.content.Intent;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void openMP3(View v){
        Intent i = new Intent(this, MP3.class);
        startActivity(i);
    }
    public void openMP4(View v){
        Intent i = new Intent(this, MP4.class);
        startActivity(i);
    }
}

File: Lab3_Kotlin/app/src/main/java/com/example/lab3_kotlin/MainActivity.kt
package com.example.lab3_kotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity(), MainFragment.OnFragmentInteractionListener {

    override fun onFragmentInteraction(result: String) {
        var fragment = supportFragmentManager.findFragmentById(R.id.resultFragment) as ResultFragment
        var fragment2 = supportFragmentManager.findFragmentById(R.id.fileFragment) as FileFragment
        if(fragment.isInLayout){
            fragment.setText(result)
        }
        if(fragment2.isInLayout){
            fragment2.setCurrentResult(result)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

}
